astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -60,12 +50,38 @@ def __init__(  # nosec: B107
         self.port = port
         self.username = username
         self.password = password
-        self.known_hosts = known_hosts
+        self.known_hosts: bytes | str = os.path.expanduser(known_hosts)
```

Comment:
Non-blocking nit: Having this be a bytes or a string doesn't feel "right" -- like what we store on `self` should probably be  either the known hosts path, or the `SSHKnownHosts` object depending on when we want/need/can parse it.

Reply:
for the host_key provided in the connection extras we're passing it as a byte string as is also accepted by the client and not a SSHKnownHosts objects imported from string.

Reply:
and os.path.expanduser(known_hosts) results to a string.

<img width="701" alt="Screenshot 2023-05-19 at 2 43 29 PM" src="https://github.com/astronomer/astronomer-providers/assets/10206082/cacc6c17-c577-4b1c-bebc-b505787b9833">